src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs

```diff
@@ -24,9 +24,9 @@
 namespace OpenTelemetry.Context.Propagation
 {
     /// <summary>
-    /// B3 text propagator. See https://github.com/openzipkin/b3-propagation for the specification.
+    /// A text map propagator for B3. See https://github.com/openzipkin/b3-propagation.
     /// </summary>
-    public sealed class B3Propagator : IPropagator
+    public sealed class B3Propagator : ITextMapPropagator
```

Comment:
Do we want and interface `ITextMapPropagator` or an abstract class `TextMapPropagator`?
Interface gives benefit of multi-inheritance, which I don't see a useful scenario here.
On the other side interface forbids us from adding optional methods in the future (will have to introduce ITextMapPropagator2, ITextMapPropagator3, ...).

Reply:
yes i was thinking.. and agree :)

Reply:
Base class seems like a good idea.

Just FYI as of C#8 you can add default implementation on an interface. Check out: https://devblogs.microsoft.com/dotnet/default-implementations-in-interfaces/

I haven't ever used it, but it's there 😄

Reply:
Makes me wondering - if we have `class Foobar: IFoo, IBar`:
1. Both `IFoo` and `IBar` decided to add a new default implementation and they share the same name 😂
2. `IFoo` used to have `IFoo.SayHello` and now `IBar` decided to add the same thing 😢

Reply:
Moved to abstract base class now. We follow same approach for others part in SDK, like base processor, exporter etc.